script.py

```
@@ -1,56 +1,79 @@
+# Run like this:
+# python script.py -i README.md -o output.html -t "My Project"
+# Output will always go in the 'dist/' folder, so just pass a filename for -o
 
-import argparse as parsing
-
-
-parser = parsing.ArgumentParser(
-    description="ReadME.md to .html writer",
-    formatter_class=parsing.ArgumentDefaultsHelpFormatter,
-)
-
-parser.add_argument(
-    "-i",
-    "--input",
-    action="store",
-    type=str,
-    required=True,
-    help="name of the input file eg. template.html",
-)
-parser.add_argument(
-    "-t",
-    "--title",
-    action="store",
-    type=str,
-    default="Progress",
-    help="title of the html page.",
-)
-parser.add_argument(
-    "-o",
-    "--output",
-    action="store",
-    type=str,
-    required=True,
-    help="the generated output file name , which is generated in dist folder. eg. template.html",
-)
+import argparse
+import subprocess
+import shutil
+import sys
+import time
+from pathlib import Path
 
+# ------------------------------
+# Parse command line arguments
+# ------------------------------
+parser = argparse.ArgumentParser(description="Convert a README/Markdown file to HTML")
+parser.add_argument("-i", "--input", required=True, help="Path to the input file, e.g., README.md")
+parser.add_argument("-t", "--title", default="Progress", help="Title of the HTML page")
+parser.add_argument("-o", "--output", required=True, help="Output filename (will be saved in dist/ folder), e.g., output.html")
 args = parser.parse_args()
 
 FILE_NAME = args.input
 TITLE = args.title
-OUTPUT_NAME = args.output
+OUTPUT_NAME = Path("dist") / Path(args.output).name  # Only take filename, ignore any folder
+
+# Make sure output folder exists
+OUTPUT_NAME.parent.mkdir(parents=True, exist_ok=True)
+
+# ------------------------------
+# Check input file exists
+# ------------------------------
+if not Path(FILE_NAME).is_file():
+    print(f"Error: The input file does not exist: {FILE_NAME}", file=sys.stderr)
```

it's better to raise an exception here, instead of sys.exit

---

okay sir, i will fix both sys.exit to raise an exception.